domain/modelagent/service/service.go

```diff
@@ -5,17 +5,45 @@ package service
 
 import (
 	"context"
+	"fmt"
 
-	"github.com/juju/errors"
 	"github.com/juju/version/v2"
 
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/machine"
 	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/watcher"
+	applicationerrors "github.com/juju/juju/domain/application/errors"
+	machineerrors "github.com/juju/juju/domain/machine/errors"
+	"github.com/juju/juju/internal/errors"
 )
 
+type ModelState interface {
```

Comment:
Add comment.

Reply:
Going to swap DB's this will get changed in the next PR.